### PR TITLE
Configure appsync group and add examples

### PR DESCRIPTION
**Pull request by turkenf**
### Description of your changes

This PR configures resources in the appsync group:

- [`aws_appsync_api_cache`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appsync_api_cache)
- [`aws_appsync_api_key`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appsync_api_key)
- [`aws_appsync_datasource`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appsync_datasource)
- [`aws_appsync_function`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appsync_function)
- [`aws_appsync_graphql_api`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appsync_graphql_api)
- [`aws_appsync_resolver`](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/appsync_resolver)

Fixes: https://github.com/upbound/provider-aws/issues/70

Note: I couldn't configure `aws_appsync_domain_name` and `aws_appsync_domain_name_api_association` I reported an issue for this: https://github.com/upbound/provider-aws/issues/150

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

aws_appsync_api_cache: https://github.com/upbound/provider-aws/actions/runs/3531760321
aws_appsync_api_key: https://github.com/upbound/provider-aws/actions/runs/3575793866
aws_appsync_datasource: https://github.com/upbound/provider-aws/actions/runs/3531797227
aws_appsync_function: https://github.com/upbound/provider-aws/actions/runs/3531806395
aws_appsync_graphql_api: https://github.com/upbound/provider-aws/actions/runs/3531816100
aws_appsync_resolver: https://github.com/upbound/provider-aws/actions/runs/3531824684

### CODE REVIEW

**Comment by turkenf** [comment]
/test-examples="examples/appsync/apicache.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/appsync/apikey.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/appsync/datasource.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/appsync/function.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/appsync/graphqlapi.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/appsync/resolver.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/appsync/apikey.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/appsync/apikey.yaml"

File: examples/appsync/apikey.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    apiIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: example
+    region: us-west-1
+---
+apiVersion: appsync.aws.upbound.io/v1beta1
+kind: GraphQLAPI
+metadata:
+  annotations:
+    meta.upbound.io/example-id: appsync/v1beta1/apikey
```
Comment (turkenh) [inline examples/appsync/apikey.yaml:20]: @turkenf, please add a pre-delete hook here that deletes APIKey first to ensure `GraphQLAPI` is not deleted before  `APIKey`. This seems to be root cause of the current uptest failure.

**Comment by turkenf** [comment]
/test-examples="examples/appsync/apikey.yaml"

**Comment by turkenf** [comment]
This PR is ready for review.

File: examples/appsync/function.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        testing.upbound.io/example-name: example
+    name: example
+    region: us-west-1
+    requestMappingTemplate: |
+      {
+          "version": "2018-05-29",
+          "method": "GET",
+          "resourcePath": "/",
+          "params":{
+              "headers": $utils.http.copyheaders($ctx.request.headers)
```
Comment (ulucinar) [inline examples/appsync/function.yaml:25]: Do we need a manual intervention here?

File: examples/appsync/function.yaml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+          }
+      }
+    responseMappingTemplate: |
+      #if($ctx.result.statusCode == 200)
+          $ctx.result.body
+      #else
+          $utils.appendError($ctx.result.body, $ctx.result.statusCode)
```
Comment (ulucinar) [inline examples/appsync/function.yaml:32]: Do we need a manual intervention here?

File: config/externalname.go [hunk 4]
```diff
@@ -1170,6 +1170,21 @@ var ExternalNameConfigs = map[string]config.ExternalName{
 	// Remediation Configurations can be imported using the name config_rule_name
 	"aws_config_remediation_configuration": config.ParameterAsIdentifier("config_rule_name"),
 
+	// appsync
+	//
+	// aws_appsync_api_cache can be imported using the AppSync API ID
+	"aws_appsync_api_cache": config.IdentifierFromProvider,
+	// aws_appsync_api_key can be imported using the AppSync API ID and key separated by :
+	"aws_appsync_api_key": config.IdentifierFromProvider,
```
Comment (ulucinar) [inline config/externalname.go:1178]:
```suggestion
	"aws_appsync_api_key": TemplatedStringAsIdentifierWithNoName("{{ .parameters.api_id }}:{{ .external_name }}"),
```

File: examples/appsync/function.yaml (hunk 2 above)
Comment (turkenf) [reply examples/appsync/function.yaml:25]: No need for manual intervention, this is handled by provider-aws.

File: examples/appsync/function.yaml (hunk 3 above)
Comment (turkenf) [reply examples/appsync/function.yaml:32]: No need for manual intervention, this is handled by provider-aws.

File: config/externalname.go (hunk 4 above)
Comment (turkenf) [reply config/externalname.go]: I have changed the externalname, thank you.

**Comment by turkenf** [comment]
/test-examples="examples/appsync/apikey.yaml"

**Comment by turkenf** [comment]
This Pr is ready for review @ulucinar

**Review by ulucinar — APPROVED** [review]
Thank you @turkenf, lgtm.

**Pull request merged**